### DISCUSSION TITLE
Customize join field fields `defaultColumns`

### DISCUSSION THREAD

**Discussion by tyteen4a03** [payloadcms/payload#8911] [Feature Requests & Ideas]
It would be great if we can specify exactly which fields will be shown in the Join Field View, akin to `defaultColumns`.

**Comment by mfilteau** [comment]
I would like that feature too!

Also:
- Make the "Type" column optional (currently added as the first column)
- Make the relationship column optional (currently added as the last column)
- When the user modifies the columns' visibility and order using the Column Selector, save the preferences.

**Comment by DanRibbens** [comment]
Config for default columns is done https://github.com/payloadcms/payload/pull/9982.

I didn't get the preferences piece done yet.

**Comment by mariovalney** [comment]
Adding to what's been discussed here, it would be helpful if `admin.defaultColumns` could override `hidden` / `disableListColumns` — for cases where a column should be visible in a sub-collection (embedded list), but not in the main collection list.

Alternatively, if `defaultColumns` isn't meant to serve that purpose (what I think it's true), it could make sense to introduce a new option like `columns`, which explicitly defines visible columns (with override priority), along with an option to prevent users from customizing the visible columns in such views.

I understand the general philosophy of allowing users to customize list views freely, but in sublist contexts, it's common to have more constrained use cases where users should only see certain fields.
